controllers/volumereplicationgroup_controller.go

```diff
@@ -112,24 +111,63 @@ func pvcPredicateFunc() predicate.Funcs {
 
 			log.Info("Update event for PersistentVolumeClaim")
 
-			// TODO: If finalizers change then deep equal of spec fails to catch it, we may want more
-			// conditions here, compare finalizers and also status.state to catch bound PVCs
-			return !reflect.DeepEqual(oldPVC.Spec, newPVC.Spec)
+			return updateEventDecision(oldPVC, newPVC, log)
 		},
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			log := ctrl.Log.WithName("pvcmap").WithName("VolumeReplicationGroup")
-
-			log.Info("delete event for PersistentVolumeClaim")
-
-			// PVC deletes are not of interest, as the added finalizers by VRG would be removed
-			// when VRGs are deleted, which would trigger the delete of the PVC.
+			// No need to reconcile VRG on pvc deletion.
+			// An update event with Nonzero deletion time
+			// would have been received to properly reconcile
+			// VRG.
 			return false
 		},
 	}
 
 	return pvcPredicate
 }
 
+func updateEventDecision(oldPVC *corev1.PersistentVolumeClaim,
+	newPVC *corev1.PersistentVolumeClaim,
+	log logr.Logger) bool {
+	const requeue bool = true
+
+	pvcNamespacedName := types.NamespacedName{Name: newPVC.Name, Namespace: newPVC.Namespace}
+	predicateLog := log.WithValues("pvc", pvcNamespacedName.String())
+	// If finalizers change then deep equal of spec fails to catch it, we may want more
+	// conditions here, compare finalizers and also status.phase to catch bound PVCs
+	if !reflect.DeepEqual(oldPVC.Spec, newPVC.Spec) {
+		predicateLog.Info("Spec changed. Requeing")
```

Comment:
(nit) Can we update log messages like so:
"Reconciling due to change in spec"
"Reconciling due to phase change", "oldPhase", ..., "newPhase", ...
"Reconciling due to finalizer updates on deletion"
"Reconciling due to deletion and existing VRG protection"

Reply:
ACK